annotated_types/__init__.py

```diff
@@ -307,3 +307,13 @@ class Predicate(BaseMetadata):
 UpperCase = Annotated[StrType, Predicate(str.isupper)]
 IsDigits = Annotated[StrType, Predicate(str.isdigit)]
 IsAscii = Annotated[StrType, Predicate(str.isascii)]
+
+
+@dataclass(**SLOTS)
+class Deprecate(BaseMetadata):
+    """Marks a parameter or field as deprecated"""
+
+    message: str
+
+
+Deprecated = Annotated[T, Deprecate(message='')]
```

Comment:
Would having `Deprecate` and `Deprecated` be confusing? Someone may import the wrong one and do `my_field: Annotated[str, Deprecated('')]` instead of `my_field: Annotated[str, Deprecate('')]`.

 I guess it will error right away so not this is not that important